src/MenuItem/ToggleableItem.php

```diff
@@ -0,0 +1,143 @@
+use PhpSchool\CliMenu\Util\StringUtil;
+
+class ToggleableItem implements MenuItem\MenuItemInterface
+{
+    /**
+     * @var callable
+     */
+    private $selectAction;
+
+    /**
+     * @var string
+     */
+    private $text = '';
+
+    /**
+     * @var bool
+     */
+    private $showItemExtra = false;
+
+    /**
+     * @var bool
+     */
+    private $disabled = false;
+
+    /**
+     * @var bool
+     */
+    private $toggled = false;
+
+    public function __construct(
+        string $text,
+        callable $selectAction,
+        bool $showItemExtra = false,
+        bool $disabled = false
+    ) {
+        $this->text          = $text;
+        $this->selectAction  = $selectAction;
+        $this->showItemExtra = $showItemExtra;
+        $this->disabled      = $disabled;
+    }
+
+    /**
+     * Execute the items callable if required
+     */
+    public function getSelectAction() : ?callable
+    {
+        return $this->selectAction;
+    }
+
+    /**
+     * Return the raw string of text
+     */
+    public function getText() : string
+    {
+        return $this->text;
+    }
+
+    /**
+     * Set the raw string of text
+     */
+    public function setText(string $text) : void
+    {
+        $this->text = $text;
+    }
+
+    /**
+     * The output text for the item
+     */
+    public function getRows(MenuStyle $style, bool $toggled = false) : array
+    {
+        $marker = sprintf("%s", $style->getMarkerToggled($this->toggled));
+
+        $length = $style->getDisplaysExtra()
+            ? $style->getContentWidth() - (mb_strlen($style->getItemExtra()) + 2)
+            : $style->getContentWidth();
+
+        $rows = explode(
+            "\n",
+            StringUtil::wordwrap(
+                sprintf('%s %s', $marker, $this->text),
+                $length,
+                sprintf("\n%s", str_repeat(' ', mb_strlen($marker)))
+            )
+        );
+
+        return array_map(function ($row, $key) use ($style, $length) {
+            $text = $this->disabled ? $style->getDisabledItemText($row) : $row;
+
+            if ($key === 0) {
+                return $this->showItemExtra
+                    ? sprintf('%s%s  %s', $text, str_repeat(' ', $length - mb_strlen($row)), $style->getItemExtra())
+                    : $text;
+            }
+
+            return $text;
+        }, $rows, array_keys($rows));
+    }
+
+    /**
+     * Can the item be selected
+     */
+    public function canSelect() : bool
+    {
+        return !$this->disabled;
+    }
+
+    public function showsItemExtra() : bool
+    {
+        return $this->showItemExtra;
+    }
+
+    /**
+     * Enable showing item extra
+     */
+    public function showItemExtra() : void
+    {
+        $this->showItemExtra = true;
+    }
+
+    /**
+     * Disable showing item extra
+     */
+    public function hideItemExtra() : void
+    {
+        $this->showItemExtra = false;
+    }
+
+    public function setToggled(bool $toggled)
```

Comment:
So, like the comment above, I think this would be better as two distinct methods `setChecked()` and `setUnchecked()`. There could also be a `toggle()` method which flips the current state.